src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala

```diff
@@ -45,18 +46,19 @@ trait ClusterComponent extends LeoComponent {
     def destroyedDate =         column[Timestamp]         ("destroyedDate",         O.SqlType("TIMESTAMP(6)"))
     def jupyterExtensionUri =   column[Option[String]]    ("jupyterExtensionUri",   O.Length(1024))
     def initBucket =            column[String]            ("initBucket",            O.Length(1024))
+    def serviceAccountKeyId =   column[Option[String]]    ("serviceAccountKeyId",   O.Length(254))
 
     def uniqueKey = index("IDX_CLUSTER_UNIQUE", (googleProject, clusterName), unique = true)
 
-    def * = (id, clusterName, googleId, googleProject, googleServiceAccount, googleBucket, operationName, status, hostIp, createdDate, destroyedDate, jupyterExtensionUri, initBucket) <> (ClusterRecord.tupled, ClusterRecord.unapply)
+    def * = (id, clusterName, googleId, googleProject, googleServiceAccount, googleBucket, operationName, status, hostIp, createdDate, destroyedDate, jupyterExtensionUri, initBucket, serviceAccountKeyId) <> (ClusterRecord.tupled, ClusterRecord.unapply)
   }
 
   object clusterQuery extends TableQuery(new ClusterTable(_)) {
 
     private final val dummyDate:Instant = Instant.ofEpochMilli(1000)
 
-    def save(cluster: Cluster, initBucket: GcsPath): DBIO[Cluster] = {
-      (clusterQuery returning clusterQuery.map(_.id) += marshalCluster(cluster, initBucket.toUri)) flatMap { clusterId =>
+    def save(cluster: Cluster, initBucket: GcsPath, serviceAccountKeyId: Option[WorkbenchUserServiceAccountKeyId]): DBIO[Cluster] = {
```

Comment:
I'm doing a similar dance with `serviceAccountKeyId` as with `initBucket`: it's stored in the database but _not_ unmarshaled to the `Cluster` case class in order to hide it from the user.